### PR TITLE
[config-plugins] refactor setting provisioning profile in pbxproj

### PR DESCRIPTION
**Pull request by dsokal**
# Why

This is the first piece for adding support for multi-target iOS projects in EAS Build.

1. Building multi-target projects requires providing EAS Build with multiple provisioning profiles (for each target). On the worker, given credentials sets for every target, we need to update `.pbxproj` with the names of the provided provisioning profiles.
2. In EAS CLI, we need to read the target name for a scheme chosen by the user.

# How

There are 4 commits in this PR:
- https://github.com/expo/expo-cli/pull/3076/commits/e515edd9fe89cf88ce3a6a34503795e7eadb5f34 moves XML.ts from `android` to `utils` so I can use that module in modules from the `ios` directory
- https://github.com/expo/expo-cli/pull/3076/commits/6ca8e8f98bce555684ce666e0cff660fe951ac9f updates the logic for setting the provisioning profile specifier (name) in `.pbxproj` - it's not setting it for the target identified by a name (used to set just for the first native target)
- https://github.com/expo/expo-cli/pull/3076/commits/a547920b91b1c92259a79819a535d7560d5b1264 adds module `Target` with a function that allows determining the main target for a particular scheme
- https://github.com/expo/expo-cli/pull/3076/commits/0e15421bfe9c40506d3831a944a282ea236c1dd9 cleans up imports in `uri-scheme` that were broken because I moved the `XML` module in the first commit

# Test Plan

- Existing tests pass.
- Added new unit tests for the `Target` module.
- Tested with changes in EAS CLI and EAS Build and was able to build a multi-target iOS project.

### CODE REVIEW

File: packages/uri-scheme/src/Android.ts [hunk 1]
```diff
@@ -1,17 +1,13 @@
-import {
-  AndroidManifest,
-  readAndroidManifestAsync,
-  writeAndroidManifestAsync,
-} from '@expo/config-plugins/build/android/Manifest';
-import * as Scheme from '@expo/config-plugins/build/android/Scheme';
-import { format } from '@expo/config-plugins/build/android/XML';
+import { AndroidConfig, XML } from '@expo/config-plugins';
```
Comment (EvanBacon) [inline packages/uri-scheme/src/Android.ts:1]: Use direct imports, otherwise the package isn't tree shaking anything.

Comment (EvanBacon) [reply packages/uri-scheme/src/Android.ts:1]: `master`:
```
508kB  build/index.js
508kB  [7432ms] - ncc 0.20.5
✨  Done in 8.12s.
```

`@dsokal/multi-target-support`:
```
   0kB  build/jetify
   1kB  build/parseJob2.js
   1kB  build/parseJob1.js
   1kB  build/parseJob.js
  25kB  build/xdg-open
2226kB  build/index.js
2254kB  [15893ms] - ncc 0.20.5
✨  Done in 16.51s.
```

**Review by EvanBacon — CHANGES_REQUESTED** [review]
need to fix uri-scheme, otherwise it looks fine.

**Review by wkozyra95 — APPROVED** [review]

**Review by wkozyra95 — APPROVED** [review]
Remember to remove pacakage.json changes before merging

**Pull request merged**